x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/AutoFollowCoordinator.java

```diff
@@ -241,10 +241,8 @@ synchronized void updateStats(List<AutoFollowResult> results) {
     }
 
     void updateAutoFollowers(ClusterState followerClusterState) {
-        final AutoFollowMetadata autoFollowMetadata = followerClusterState.getMetadata().custom(AutoFollowMetadata.TYPE);
-        if (autoFollowMetadata == null) {
-            return;
-        }
+        final AutoFollowMetadata autoFollowMetadata = followerClusterState.getMetadata()
+            .custom(AutoFollowMetadata.TYPE, AutoFollowMetadata.EMPTY);
 
```

Comment:
Can we bail out early here if `autoFollowMetadata.getPatterns().isEmpty() && this.autoFollowers.isEmpty()`? Since this runs on every cluster change it would be nice to have a bail out.